### ISSUE TITLE
Build repository missing configure script and associated files

### ISSUE THREAD

**Issue by swhobbit** [Fish-Git/hyperion#111]
The repository has a `configure.ac` script, but no `configure` script.  The former requires _every_ person who downloads the repository to install `automake`, which is contrary to standard practice.  

Expected behavior:

`automake` is run after changes which affect `configure`, and the generated file is committed to the repository

**Comment by Fish-Git** [comment]
Thanks Drew! I'll get it fixed as soon as I can. Thanks for reporting it.

**Comment by Fish-Git** [comment]
Closed by commit 865a48e00b2f4282c03c29669d3278237b3e8d26.

**Comment by swhobbit** [comment]
It appears not all the files normally generated by `automake` were added.

```
configure:                                           
configure:  Package destination directory prefixes:  
configure:                                           
configure:     Libraries:  $(libdir)/$(PACKAGE)             
configure:     Data:       $(datadir)/hercules      
configure:     Locale:     ${datarootdir}/locale              
configure:                                           
configure: creating ./config.status
config.status: error: cannot find input file: Makefile.in
[marvin-the-martian:0,~/src/hyperion,04:43:10,125] ls -l Makefile.* 
-rw-r--r-- 1 ahd ahd 27654 Jul  7 16:40 Makefile.am
```

**Comment by swhobbit** [comment]
Hmmm ... how do I reopen this?

**Comment by Fish-Git** [comment]
> Hmmm ... how do I reopen this?

Well I have a nice big _"Reopen and comment"_ button next to my big green _"Comment"_ button on my screen, but I'm guessing you don't see one of those on your screen, do you? I guess that means only I, the repository owner, can re-open GitHub Issues. Bummer. That kind of sucks, doesn't it? Oh well.

I guess I'll go ahead and re-open this issue then, since, based on your comment, the problem has _obviously_ not been fixed yet.&nbsp;`:(`

(issue re-opened)

_**QUESTION:**_ What files do I need to add to the repository besides the `configure` script you originally mentioned and the `Makefile.in` file you mentioned just now? Are there any others? **Note:** _I know very little about *Nix autoconf!_ `(*)`
&nbsp;
____
`(*`)&nbsp;&nbsp;&nbsp;Only enough to get by!&nbsp;&nbsp;&nbsp;I mostly just do a bunch of Googling and then "fake it" as best I can.

**Comment by swhobbit** [comment]
> Only enough to get by!   I mostly just do a bunch of Googling and then "fake it" as best I can.

I'm not just anyone except the authors do more, and for all I know *they* fake it.

Just run automake on a fresh clean download of the repository, and add any files it creates like Makefile.in.  (What did you feed automake to get it to run to completion?)

**Comment by Fish-Git** [comment]
> (What did you feed automake to get it to run to completion?)

We have a script called `autogen.sh` that runs `aclocal`, `autoheader`, `automake` and `autoconf`. An excerpt of the pertinent portion is as follows:
```
echo $ECHO_N "aclocal...    $ECHO_C" && aclocal -I m4 -I autoconf >>./autogen.log 2>&1 && echo "OK.  (25% done)" &&
echo $ECHO_N "autoheader... $ECHO_C" && autoheader                >>./autogen.log 2>&1 && echo "OK.  (50% done)" &&
echo $ECHO_N "automake...   $ECHO_C" && automake --add-missing    >>./autogen.log 2>&1 && echo "OK.  (75% done)" &&
echo $ECHO_N "autoconf...   $ECHO_C" && autoconf                  >>./autogen.log 2>&1 && echo "OK.  (100% done)"
```
&nbsp;
After running `autogen.sh` on a clean clone and comparing the resulting directory tree with the original tree, it appears the files that end up being created by `autogen.sh` are:
```
aclocal.m4
autogen.log
autom4te.cache
config.h.in
Makefile.in

autom4te.cache/output.0
autom4te.cache/output.1
autom4te.cache/requests
autom4te.cache/traces.0
autom4te.cache/traces.1

html/Makefile.in
m4/Makefile.in
man/Makefile.in
util/Makefile.in
```

Except for `autogen.log` (which is the log file created by `autogen.sh` itself), should _**all**_ of the above be added?

Or should _only_ the `Makefile.in` files be added?

What about the files in the `autom4te.cache` directory for example? (and the `autom4te.cache` file?) Should those be added too? I'm guessing _no_, but I'm not sure, which is why I'm asking!

Any enlightenment/guidance would be greatly appreciated.

Thanks!

**Comment by Fish-Git** [comment]
@ivan-w: I got your phone message you left me the other day but I didn't call you back because it unfortunately costs me  a small fortune to call overseas.

You stated in your message a concern regarding this particular issue (that you for some reason felt the `configure` script should in fact _not_ be added to the repository). Can you state publicly what your concern is? (i.e. why you feel that way?) Or would you rather your concern remain private? _(I'm just a little bit confused by your concern!)_

**Comment by swhobbit** [comment]
@ivan-w,

Add the `configure` script is a standard practice.  For example, from packages I've got on my desktop:

./dhcp-4.3.5/configure
./stunnel-5.42/configure
./suite3270-3.6/configure

If there is a problem with including `configure` in Hercules, Hercules's problem needs to be fixed, not drop the script. If everyone had to run `automake` (_Ceiling Cat help them_), it would be redundant for `configure` to exist.

People should NOT need to include `automake` to configure and build a package.  Running `configure` should be sufficient.

**Comment by swhobbit** [comment]
The lack of an **obvious** top level read me is part of this issue.  There's no "build it this way" in a directory with 375 files ... :-)

**Comment by Fish-Git** [comment]
@swhobbit,

FYI: still waiting for an answer:

> Except for autogen.log (which is the log file created by autogen.sh itself), should all of the above be added?
> 
> Or should only the Makefile.in files be added?
> 
> What about the files in the autom4te.cache directory for example? (and the autom4te.cache file?) Should those be added too? I'm guessing no, but I'm not sure, which is why I'm asking!
> 
> Any enlightenment/guidance would be greatly appreciated.

**Comment by swhobbit** [comment]
_(he thinks **I** know? snicker!)_

I was say add anything with `.in` suffix. header files, Makefiles, and the like.

**Comment by Fish-Git** [comment]
> (he thinks I know? snicker!)

Well **_I_** sure as hell don't!&nbsp;`:)`

> I would say add anything with a .in suffix (header files, Makefiles, and the like).

What you suggest sounds perfectly reasonable to me so that's what I guess I'll do. Give me a few minutes or possibly another day to get it done though. It's getting late here and I'm getting tired. Give it a try again once you see the commit hit the repository.

**Comment by swhobbit** [comment]
> Well I sure as hell don't! :)
GMTA

**Comment by Fish-Git** [comment]
Fixed _(hopefully!)_ by commit aedeb54329361f9d43de21ac285b39e8572d8c75.

@swhobbit: please retry and report back. I won't close this issue until I'm sure it's fixed.

**Comment by swhobbit** [comment]
`configure` runs successfully.

**Comment by Fish-Git** [comment]
> configure runs successfully.

Fantastic! Am closing this issue as resolved.

Thanks for your help, Drew!

**Comment by ivan-w** [comment]
Hey guys !

Personally, I think it's a bit of a mistake to have "configure" (and all the necessary auto magic generated bits) in the GIT source repository... GIT is suppose to be a *source* repository - whereas "configure" and whatever happens to exist after you've run the necessary auto magic is *generated* (hence not *source* per-se).
I understand it is more convenient, but it also generates a lot of garbage, making it difficult to track whenever a bug is inadvertently introduced.. And possibly generating 10K+ lines patches even if one changes a single character in say configure.ac (or maybe happens to have run autogen.sh with a different version of the automagic tools than the previous one who committed something).
The solution would be to have an automated snapshot process to generate the 'dist' package (the output of "make dist") - which generates a tarball with all the necessary bit to run "configure && make && make install".

I won't sweat over it though.. Just food for thought !

--Ivan

**Comment by Fish-Git** [comment]
Hmmm... I see your point, Ivan: a source code repository is not the same thing as a distribution tarball. Distribution tarballs (output of `make dist` ) are the "things" which normally include the `configure` script and all of the `*.in` files (`config.h.in`, `Makefile.in`, etc). The repository should, as you say, only contain the product's _source code_, but not any _distribution_ components.

That makes perfect sense and was something I hadn't thought about before! Thanks!

_**However...**_ (you just _knew_ there was going to be a "however", didn't you?), I don't believe the "problem(s)" (drawback(s)) that you listed are serious enough or likely to occur frequently enough (IMO) to warrant undoing what was just done.

If a user has different automagic tools than the ones I used to generate the `configure` and `*.in` files with, it: 1) shouldn't matter, and 2) would only indicate either the version they have is broken or vice versa, with the "fix" for such a situation being either: a) tell the other user to use the provided `configure` script instead of trying to generate their own, or: b) me updating the version(s) of my own automagic tools to use the newer version(s) the other person presumably has.

In other words, I personally don't feel your reasoning for _not_ including them in the repository is _compelling_ enough! It just seems to me that you're grasping at straws to find a reason for _not_ including them, and the reason you provide, as I said, doesn't (to me!) seem very likely to occur (or if it does, not all that difficult to overcome). The "problem" you describe doesn't (again, to _me;_ others may fee differently!) seem "severe" or "serious" enough to warrant _not_ including them.

And including them makes it a _heck_ of a lot easier for our users dabbling in Hercules development, which IMO _outweighs_ your reasoning for _not_ including them.

Sorry!&nbsp;`:(`

But _thank you for your thoughts!_ It's great feedback, and _that_ is _**very**_ much appreciated!&nbsp;`:)`

**Comment by ivan-w** [comment]
I think we had this discussion in 2003 with I believe Ian Thornton .. and the conclusion was that the output of a generated object shouldn't be included.. 

It's your fork.. your decision.. I disagree with your decision, but your call..

--Ivan

**Comment by ivan-w** [comment]
And contrary to Drew's point of view, I believe automake/autoconf etc.. are as needed as gcc, gmake and whatnot when building a source pulled project... (Again, my point of view)

**Comment by Fish-Git** [comment]
> I think we had this discussion in 2003 with I believe Ian Thornton .. and the conclusion was that the output of a generated object shouldn't be included..

I don't remember that.

> It's your fork.. your decision.. I disagree with your decision, but your call..

I'm just trying to listen to my "customers" so to speak. It's not causing you any problems, is it?

**Comment by Fish-Git** [comment]
Note: This issue is essentially a Duplicate of issue #122.

**Comment by mhoes** [comment]
> Note: This issue is essentially a Duplicate of issue #122.

Note: Not really, the **resolution** to this issue #111 is the **cause** of issue #122, which sounds similar but is strictly speaking not the same thing as a 'duplicate'. ;)

**Comment by Fish-Git** [comment]
> > Note: This issue is essentially a Duplicate of issue #122.
> 
> Note: Not really, the resolution to this issue #111 is the cause of issue #122, which sounds similar but is strictly speaking not the same thing as a 'duplicate'. ;)

Point made.&nbsp;`:)`
